### ISSUE TITLE
Update failed: error sending request for url: operation timed out

### ISSUE THREAD

**Issue by uc-asa** [shorebirdtech/updater#174]
Hey team,

We are using shorebird for patch and we notice It's impacting our splash time.

So when user is try to launch the app the update checker request to check new update and It's keep waiting until request resolve.
And that lend to stuck in splash screen for more than 20 seconds. It's very bad experience in terms of user prospective.
 Here is logcat


`updater::updater: Update failed: error sending request for url (https://api.shorebird.dev/api/v1/patches/check): operation timed out
                                                                                                    
                                                                                                    Caused by:
                                                                                                        operation timed out
2024-05-28 15:18:08.394  2849-2964  flutter                 com.flashmonk.flash                  I  updater::updater: Update thread finished with status: Update had error
2024-05-28 15:18:08.601  2849-2997  TRuntime.C...ortBackend com.flashmonk.flash                  E  Couldn't open connection, returning with 500
2024-05-28 15:18:08.612  2849-2997  Compatibil...geReporter com.flashmonk.flash                  D  Compat change id reported: 194532703; UID 10932; state: ENABLED
2024-05-28 15:18:28.654  2849-3005  flutter                 com.flashmonk.flash                  E  updater::c_api: Error checking for update: error sending request for url (https://api.shorebird.dev/api/v1/patches/check): operation timed out
                                                                                                    
                                                                                                    Caused by:
                                                                                                        operation timed out
2024-05-28 15:18:28.655  2849-2849  flutter                 com.flashmonk.flash                  W  updater::config: Updater already initialized, ignoring second shorebird_init call.`



I would need solution here. Do we have an option to avoid sync call?.

**Comment by uc-asa** [comment]
My app id is different and it's 42fc4f05-2fde-4894-8132-f2d719e4f217
Also we are using flutter 3.19.5 version to build.


I could there are some ANR related also is getting fixed with flutter version 3.22.0. Is there any specific reason?.
Most of them issues were due to shorebird library. Also are you sure will this issue will go away?.

Or we should stop using shorebird?. As we are on team plan we shouldn't expecting this kind of impact on our app.

**Comment by uc-asa** [comment]
Okay, we might update the flutter version. But I've question If it has been update inside shorebird shouldn't be notified or updated to the shorebird flutter library as new version so at least whoever is using these services they can update or perform respective steps to solve this.

Also I've another question, It started happening very recently do we have some date or flutter version since when it's doing for better tracking.

**Comment by eseidel** [comment]
> Okay, we might update the flutter version. But I've question If it has been update inside shorebird shouldn't be notified or updated to the shorebird flutter library as new version so at least whoever is using these services they can update or perform respective steps to solve this.

I'm not sure I fully understand the question.  We definitely could do more to notify customers when we make changes, yes.   We don't currently post on a mailing list, but we do announce changes on:
Discord: https://discord.gg/shorebird
Twitter: https://twitter.com/shorebirddev
GitHub: https://github.com/shorebirdtech/shorebird/releases
https://github.com/shorebirdtech/shorebird/blob/main/RELEASE_NOTES.md

But I think we could definitely do more.

> Also I've another question, It started happening very recently do we have some date or flutter version since when it's doing for better tracking.

My understanding of the issue is that it's triggered by having a shorebird update already running at the time that the app is launched.  Previously (before we fixed the hang) if shorebird_init was called (a function called during Flutter's boot) and there was already an update going, shorebird_init would hang until the update completed.  We fixed the updater library to no hang like this in our Flutter 3.22.0 release.  Things that could have caused that would be adoption of a package like Firebase Cloud Messaging in your application (that launches another copy of Flutter in the background).
